sklearn/preprocessing/label.py

```diff
@@ -732,7 +732,7 @@ def fit_transform(self, y):
         class_mapping = np.empty(len(tmp), dtype=dtype)
         class_mapping[:] = tmp
         self.classes_, inverse = np.unique(class_mapping, return_inverse=True)
-        yt.indices = inverse[yt.indices].astype(yt.indices.dtype, copy=False)
```

Comment:
you could use `np.asarray(..., dtype=)` to more closely reflect this operation, I think